nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol

```diff
@@ -89,94 +119,370 @@ contract WorkLock {
         escrow = _escrow;
         startBidDate = _startBidDate;
         endBidDate = _endBidDate;
+        endCancellationDate = _endCancellationDate;
         boostingRefund = _boostingRefund;
         stakingPeriods = _stakingPeriods;
+        minAllowedBid = _minAllowedBid;
+        maxAllowableLockedTokens = escrow.maxAllowableLockedTokens();
+        minAllowableLockedTokens = escrow.minAllowableLockedTokens();
+        creator = msg.sender;
     }
 
     /**
     * @notice Deposit tokens to contract
     * @param _value Amount of tokens to transfer
     */
     function tokenDeposit(uint256 _value) external {
-        require(block.timestamp <= endBidDate, "Can't deposit more tokens after end of bidding");
+        require(block.timestamp < endBidDate, "Can't deposit more tokens after end of bidding");
         token.safeTransferFrom(msg.sender, address(this), _value);
         tokenSupply += _value;
         emit Deposited(msg.sender, _value);
+        require(tokenSupply / minAllowableLockedTokens <= uint128(0) - 1, "Potential overflow for bidder index");
```

Comment:
..."Potential" overflow?  What is this condition indicating?

Reply:
It means that possible situation when number of bidders will be so big and will be overflow in indexes. I know it's not realistic but prefer to check such things

Reply:
I appreciate your dislike of undefined behavior :-)